internal/ecies_params_ethereum.go

```diff
@@ -107,29 +86,15 @@ var (
 )
 
 var paramsFromCurve = map[elliptic.Curve]*ECIESParams{
-	ethcrypto.S256(): ECIES_AES128_SHA256,
```

Comment:
Potentially, "The Ethereum's S256 (which corresponds to secp256k1)" could have been used by some customer that generated their device certificate "offline" (and signed with offline CA) without `lmp-dev-reg`. I think is is very unlikely though.

Reply:
If we wanted to we could use: https://github.com/decred/dcrd/tree/master/dcrec/secp256k1 to bring it back in without relying on go-ethereum. It has the copyfree ISC license.

Reply:
Unfortunately, the decred above uses the go-ethereum under the hood, as do all other 3PP libraries implementing this algorithm except the https://github.com/ThePiachu/Golang-Koblitz-elliptic-curve-DSA-library/blob/master/bitelliptic/bitelliptic.go#L326. But that was not maintained for the last 8 years and is not even compatible with the Golang's crypto interfaces.

So, initially I wrote the code to support the S256 by copying the curve numbers from the spec and using reflection to create internal Golang crypto objects. And it worked well (I could encrypt/decrypt data in go-ethereum compatible way. I still have that code around, but it is enough of a contemporary programmer's nightmare to not share it anywhere.

But, the truth is: no single customer could ever use the S256 curve, as it is not supported by the Golang's TLS implementation. Such certificates would simply fail as "not supported EC algorithm" when reading such client certificates.

I thuought I was enough explicit about that in the commit message. Please, let me know if you read it and it is not clear enough.

Reply:
In the case of an offline device certificate we need to make sure that both ota-lite and device-gateway won't accept it.

Reply:
i think i just copied this file as close to "as-is" as possible at the time. I wasn't sure how much it might change over time and just wanted to make sure it would be easy to pull in changes. Turns out there have never been changes.

Reply:
> In the case of an offline device certificate we need to make sure that both ota-lite and device-gateway won't accept it.

I don't think so.

There are 100500 ways how a user could craft a particular certificate in a way to make it pass the ota-lite validation but fail the TLS connection. But, as long as the device-gateway will not accept a TLS connection with some client certificate - that certificate is deemed unsupported from our perspective.

By generating the device certificate using a custom tool (not vanilla lmp-device-register), the customer steps into a danger zone; so they are responsible to verify that their solution works end-2-end.

Reply:
> I thought I was explicit about that in the commit message. Please, let me know if you read it and it is not clear enough.

Actually my apologies, I forgot to read that first. That message is very clear and perfect.